rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -170,6 +175,16 @@ pub async fn reconcile(spark_application: Arc<SparkApplication>, ctx: Arc<Ctx>)
 
     let client = &ctx.client;
 
+    // Fix for #457
+    // Skip reconcyling the SparkApplication if it has a non empty status.
+    if spark_application.status.is_some() {
```

Comment:
I think it makes sense to move this to a named function on `SparkApplication` with some docs: 
```rust
impl SparkApplication {
    /// Returns if this [`SparkApplication`] has already created a Kubernetes Job doing the actual `spark-submit`.
    ///
    /// This is needed because Kubernetes will remove the succeeded Job after some time. When the spark-k8s-operator is
    /// restarted it would re-create the Job, resulting in the Spark job running multiple times. This function assumes
    /// that the [`SparkApplication`]'s status will always be set when the Kubernetes Job is created. It therefore
    /// checks if the status is set to determine if the Job was already created in the past.
    ///
    /// See the bug report [#457](https://github.com/stackabletech/spark-k8s-operator/issues/457) for details.
    pub fn k8s_job_has_been_created(&self) -> bool {
        self.status
            .as_ref()
            .map(|s| !s.phase.is_empty())
            .unwrap_or_default()
    }
}
```

I did change the actual check to check the status is not empty, but I guess it's also ok to keep the current `status.is_some()` check

Reply:
Done